спринта №15 #1

Gammius opened the pull request:
спринта №15 #1

Gammius committed:
спринта №15

i-masloed commented on app/src/main/java/com/example/playlist_maker/Creator.kt:
```diff
@@ -0,0 +1,30 @@
+        return TrackInteractorImpl(getTrackRepository())
+    }
+
+    fun provideSearchHistoryInteractor(context: Context): SearchHistoryInteractor {
```
Рекомендация: было бы здорово  не получать контекст из аргументов метода( в этом случае - на ui-слое в активити), а при старте приложения в App добавить контекск в Creator:

```
//App
class App:Application(){

override fun onCreate() {
        super.onCreate()
        Creator.initialize(applicationContext)
}

//Creator
object Creator {

private lateinit var context: Context

    fun initialize(ctx: Context){
        context = ctx
    }

   private fun getSearchHistoryRepository(): SearchHistoryRepository {
        return SearchHistoryRepositoryImpl(context)
    }



}

```

Gammius committed:
спринта №15
исправление критических ошибок
по некретичемкой: Студия говорит об утечки памяти(пока не разобрался)

Gammius committed:
спринта №15
вынес метод isDarkThemeSaved() в интерфейс репозитория
убрал явные приведения типов в активити и в SettingsInteractorImpl
